spec/models/loan_spec.rb

```diff
@@ -0,0 +1,83 @@
+require 'rails_helper'
+
+describe Loan, :type => :model do
+  it_should_behave_like 'translatable', ['Description', 'ShortDescription']
+  it_should_behave_like 'mediable'
+
+  it 'has a valid factory' do
+    expect(create(:loan)).to be_valid
+  end
+
+  context 'model methods' do
+    let(:loan) { create(:loan) }
+
+    describe '.name' do
+      context 'with cooperative' do
+        it 'uses cooperative name' do
+          expect(loan.name).to eq I18n.t :project_with, name: loan.cooperative.name
+        end
+      end
+
+      context 'without cooperative' do
+        let(:loan) { create(:loan, cooperative: nil) }
+        it 'uses project id' do
+          pending('is not currently possible given DB constraints')
+          expect(loan.name).to eq I18n.t :project_id, id: loan.id.to_s
+        end
+      end
+    end
+
+    describe '.country' do
+      it 'gets country' do
+        expect(loan.country).to eq Country.where(name: loan.division.super_division.country).first
+      end
+
+      context 'without country' do
+        before do
+          @division = create(:division)
+          @loan = create(:loan, source_division: @division.id)
+          @us = create(:country, name: "United States")
+        end
+        it 'gets united states' do
+          expect(@loan.country).to eq @us
+        end
+      end
+    end
+
+    describe '.location' do
+      it 'returns city and country' do
+        expect(loan.location).to eq "#{loan.cooperative.city}, #{loan.country.name}"
```

Comment:
Here again I'd prefer to explicitly set the city and country when creating the loan and then check that the text matches the expectation.